File: internal/models/personnel.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// DutyPersonnel maps to public.duty_personnel
type DutyPersonnel struct {
	ID                 uuid.UUID  `json:"id"`
	StationID          *uuid.UUID `json:"station_id,omitempty"`
	FleetID            *uuid.UUID `json:"fleet_id,omitempty"`
	FullName           string     `json:"full_name"`
	Rank               string     `json:"rank"`
	Designation        *string    `json:"designation,omitempty"`
	Shift              *string    `json:"shift,omitempty"`
	DutyStatus         string     `json:"duty_status"`
	IsStationCommander bool       `json:"is_station_commander"`
	NFCTagID           *string    `json:"nfc_tag_id,omitempty"`
	PinCode            *string    `json:"pin_code,omitempty"`
	CreatedAt          time.Time  `json:"created_at"`
	UpdatedAt          time.Time  `json:"updated_at"`
}

// CreatePersonnelRequest is the request body for creating a new personnel record
type CreatePersonnelRequest struct {
	StationID          *uuid.UUID `json:"station_id"`
	FleetID            *uuid.UUID `json:"fleet_id"`
	FullName           string     `json:"full_name" binding:"required"`
	Rank               string     `json:"rank" binding:"required"`
	Designation        *string    `json:"designation"`
	Shift              *string    `json:"shift"`
	DutyStatus         string     `json:"duty_status"`
	IsStationCommander bool       `json:"is_station_commander"`
	NFCTagID           *string    `json:"nfc_tag_id"`
	PinCode            *string    `json:"pin_code"`
}

// UpdateDutyStatusRequest is the request body for updating duty status
type UpdateDutyStatusRequest struct {
	DutyStatus string `json:"duty_status" binding:"required"`
}

File: frontend/src/api/personnel/personnel.js
/**
 * api/personnel.js
 *
 * Duty personnel API calls. Maps to:
 *   GET    /api/v1/personnel
 *   GET    /api/v1/personnel/:id
 *   POST   /api/v1/personnel
 *   PATCH  /api/v1/personnel/:id/duty-status
 */

import api from '../client/client';

export const personnelApi = {
  list: () => api.get('/personnel'),
  getById: (id) => api.get(`/personnel/${id}`),

  /**
   * @param {{ full_name: string, rank: string, station_id?: string, nfc_tag_id?: string, pin_code?: string, shift?: string }} data
   */
  create: (data) => api.post('/personnel', data),

  /**
   * @param {string} id
   * @param {object} data — partial update fields
   */
  update: (id, data) => api.put(`/personnel/${id}`, data),

  /**
   * @param {string} id
   * @param {'On Duty' | 'Off Duty' | 'On Leave'} status
   */
  updateDutyStatus: (id, status) =>
    api.patch(`/personnel/${id}/duty-status`, { duty_status: status }),
};
